op-service/sources/l1_beacon_client.go

```diff
@@ -298,9 +298,13 @@ func blobsFromSidecars(blobSidecars []*eth.BlobSidecar, hashes []eth.IndexedBlob
 		}
 
 		if !skipBlobVerification {
-			// confirm blob data is valid by verifying its proof against the commitment
-			if err := eth.VerifyBlobProof(&sidecar.Blob, kzg4844.Commitment(sidecar.KZGCommitment), kzg4844.Proof(sidecar.KZGProof)); err != nil {
-				return nil, fmt.Errorf("blob at index %d failed verification: %w", i, err)
+			// recompute the commitment from the blob data and check it matches the commitment from the sidecar
+			recomputedCommitment, err := kzg4844.BlobToCommitment(sidecar.Blob.KZGBlob())
+			if err != nil {
+				return nil, fmt.Errorf("cannot compute KZG commitment for blob %d: %w", i, err)
+			}
+			if recomputedCommitment != kzg4844.Commitment(sidecar.KZGCommitment) {
+				return nil, fmt.Errorf("recomputed commitment %s does not match expected commitment %s", recomputedCommitment, kzg4844.Commitment(sidecar.KZGCommitment))
```

Comment:
This looks right to me, but it makes the invariant we're looking for transitive instead of explicit. It enforces that:

* `ih.Hash == KZGToVersionedHash(sidecar.KZGCommitment)` (L296)
* `sidecar.KZGCommitment == BlobToCommitment(sidecar.Blob)` (L306)

But we don't actually care about the value of `sidecar.KZGCommitment` that the beacon node has given us now that we don't have `sidecar.KZGProof` to use as a shortcut for verifying the blob. We just want to ensure that `ih.Hash == KZGToVersionedHash(BlobToCommitment(sidecar.Blob))`.

Reply:
Totally agree. I've refactored it to not use the "metadata" from the beacon API at all -- it now only uses the blob data. We anticipate that the beacon API will soon be changed to only return that data anyway. [830e96d](https://github.com/ethereum-optimism/optimism/pull/17725/commits/830e96dfdda0d2cfb0e7f7f3d598f79254f1a4bb)